File: python/omniversion/dependency/audit/advisory.py
#!/usr/bin/env python
from dataclasses import dataclass
from typing import Optional
from ...pretty import pretty


@dataclass
class Advisory:  # pylint: disable=too-many-instance-attributes
    access: Optional[str]
    cvss_score: Optional[float]
    identifier: Optional[int]
    overview: Optional[str]
    patched_versions: Optional[str]
    recommendation: Optional[str]
    references: Optional[str]
    severity: Optional[str]
    title: Optional[str]
    url: Optional[str]
    vulnerable_versions: Optional[str]

    def __str__(self):
        severity = pretty.severity(self.severity)
        version = pretty.bright_on_lightblack(self.patched_versions)
        return f"a known vulnerability of severity {severity} and should be updated to {version}"

File: cli/cmd/parse/npm/types/advisory.go
package types

type NpmAdvisory struct {
	Findings []struct {
		Version string
		Paths   []string
	}
	VulnerableVersions string `json:"vulnerable_versions"`
	ModuleName         string `json:"module_name"`
	Severity           string
	Access             string
	PatchedVersions    string `json:"patched_versions"`
	CVSS               struct {
		Score        float64
		VectorString string
	}
	Recommendation string
	Id             int
	References     string
	Title          string
	Overview       string
	Url            string
}
